### ISSUE TITLE
Read sometimes not cached in functional tests.

### ISSUE THREAD

**Issue by KamilLepek** [Open-CAS/ocf#212]
[LOGS from pyocf](https://s3.us-east-2.amazonaws.com/open-cas-logs/index.html?project=ocf&pr=199&category=functional-tests&run=305)
Test name: test_start_read_first_and_check_mode

I encountered this several times, however it is very sporadic. Seems like it isn't related to hardware configuration.

**Comment by imjfckm** [comment]
If OCF encounters a read miss it:
1. Reads data from core device
2. Copies over data to separate buffer
3. Completes original IO to exported object with original data
4. Inserts data copied in step 2 into cache

The IO is completed at step 3, execution of test continues and assertion fails because step 4 wasn't executed yet.

We gotta think of some way to wait for backfill or drop assertions which check each and every time if data that should be inserted is inserted. Maybe split it out to dedicated tests which make sure bf run, and don't bother with it in other tests?

**Comment by arutk** [comment]
We plan to implement a utility to settle OCF to make sure there is no outstanding activity after I/O is completed. Initial version of this feature is available here:
https://github.com/Open-CAS/ocf/pull/588/commits/4d7d968898d4f6435c91752109daf556fb5b1b32
Once the settle mechanism is in place, we'll be able to update tests and fix this issue.
